File: back-end/src/main/java/es/ull/project/application/usecase/operation/OperationDeleteUseCase.java
package es.ull.project.application.usecase.operation;

import java.util.Optional;

import es.ull.project.domain.valueobject.Operation;
import es.ull.project.domain.valueobject.OperationName;

public interface OperationDeleteUseCase {
    
    /**
     * Delete an operation
     * 
     * @param operation
     * @return operation deleted
     */
    Optional<Operation> deleteOperation(OperationName operationName);
}

File: front-end/src/application/usecase/operation/operation-delete-use-case.ts
import { UllUUID } from '@ull-tfg/ull-tfg-typescript';
import { Operation } from '../../../domain/value-object/operation';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import { OperationName } from '@/domain/value-object/operation-name';

export interface OperationDeleteUseCase {
    /**
     * Delete an operation
     * @param operationId
     * @returns operation deleted
     */
    deleteOperation(name: OperationName): Promise<Either<DataError, Operation>>;
}
